src/libertem/analysis/base.py

```diff
@@ -154,7 +154,26 @@ def __iter__(self):
         return iter(self.results)
 
 
-class Analysis:
+class AnalysisRegistry(type):
+
+    registry = {}
+
+    def __init__(cls, name, args, bases):
+
+        analysis_info = cls.get_analysis_info()
+
+        if analysis_info is None:
+            return
+        cls.registry[analysis_info['type']] = {
+            "class": cls,
+        }
+
+    @classmethod
+    def get_analysis_by_type(cls, type_):
+        return cls.registry[type_]
+
+
+class Analysis(metaclass=AnalysisRegistry):
```

Comment:
...as soon as it is clear what this does and why, we should discuss if there is maybe a way to achieve the same goal with more common programming patterns, CC @sk1p .

Reply:
The alternative is not as elegant, that would be explicit registration (`Aanlysis.register('SOME_NAME', SomeAnalysisClass)`), or a central list of all available analyses, like we have for the `DataSet`s. This registry pattern is actually one of the most common metaclass uses, and is often used as an example in tutorials and books. I also use it for the registry of download formats, for example.

Reply:
Ok, in that case a comment pointing to a concise explanation would be good! :-)

Reply:
There is actually a new feature available since Python 3.6, which is the `__init_subclass__` method. That can be used as a shorter alternative, and we can skip the metaclass explanations: https://stackoverflow.com/a/50099920/540644

Still not a very well known feature, but a bit less code needed, and possibly more straight-forward to understand. If we decide on that version, the download format registry should also be changed, so we have some consistency.

Reply:
> (...) `__init_subclass__` method (...)

Oh, *that* looks neat and straight to the point. Much more intuitive. :+1:

Reply:
So I will change it :)